### PR TITLE
[Bugfix] Reduce GitHub Action Permissions (#2)

### LINKED ISSUE #385: [Bug] GitHub Action Permissions
At the moment, your GitHub Actions do not have any specialised permission settings.  Hence, GitHub grants a default set of permissions to the session token which does not always fit the intention of the actual workflows.

Please consider to reduce the permissions you grant in order to prevent unintended side effects.

For further reading, please have a look at both https://github.com/citation-file-format/cffconvert-github-action/issues/67 and the official GitHub documentation concerning this topic ("Security hardening for GitHub Actions").

### PR DESCRIPTION
**Pull request by kevinmatthes**
**Related issues**

Refs:  fixes #385

**Describe the changes made in this pull request**

I reduced the GitHub Action permissions in order to enhance the security of the project.  Especially for the CFF validation Action, there are even more paranoid settings possible.  From my experiences when setting up CFF for other projects, these are sometimes rejected by the maintainers.  Thus, I left them out for this Pull Request, initially.  If you would like to have them activated, please tell me.

**Review checklist**

- [ ] Please check if the pull request is against the correct branch  
(format/schema/semantic documentation changes: `develop`; typos, meta files, etc.: `main`)
- [ ] Please check if all changes are recorded in `CHANGELOG.md` and adapt if necessary.
- [ ] Please run tests locally.
```bash
cd $(mktemp -d --tmpdir cff.XXXXXX)
git clone https://github.com/citation-file-format/citation-file-format .
git checkout <branch>
python3 -m venv env
source env/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pytest
```

- [ ] Please validate the correctness of the GitHub Actions.  Some further permissions might be required.  If so, please tell me such that I can fix it.

### CODE REVIEW

**Comment by kevinmatthes** [comment]
As I wrote in the description, the GitHub Actions might require further permissions than just reading access to the file contents.  I consider it unlikely but it might happen that some of the Actions fail due to too few permissions.  I know that the settings for the CFF validation work with just reading access, so this Action should not fail.

From all repositories I set CFF up for (and whose maintainers accepted CFF being set up), one project refused the extremely paranoid settings for the CFF validation, this is pinning the Actions to their respective revision numbers.  If I shall change this for any Action (activate the pinning to the revision number), please tell me.

I also added the GitHub manual page named in #385 to the `CITATION.cff` as a new reference.  I shelved it in as a blog entry.  Other options would have been manual or website.  The blog entry appeared to be the ideal balance, at least to me.  I can change it, if necessary.

**Comment by sdruskat** [comment]
Thanks, @kevinmatthes, for this pull request, and sorry for taking such a long time with it.

I think it makes a lot of sense to tighten restrictions, and would be happy to merge this as is (I don't think pinning the actions is something we strictly need for now). What do you think @jspaaks?

As for adding the GitHub guide to the CFF file, this seems fair, as the work (i.e., this repository) would build on it after merge.

File: CITATION.cff [hunk 1]
```diff
@@ -110,3 +110,11 @@ references:
     type: proceedings
     volume: 27
     year: 2006
+  - abstract: Good security practices for using GitHub Actions features.
+    date: 2022-09-14
+    date-accessed: 2022-09-20
+    editors:
+      - name: GitHub, Inc.
```
Comment (jspaaks) [inline CITATION.cff:117]:
```suggestion
  - authors:
      - name: GitHub, Inc.
    abstract: Good security practices for using GitHub Actions features.
    date-published: 2022-09-14
    date-accessed: 2022-09-20
```
`authors` is required, renamed `date` to `date-published`

**Comment by jspaaks** [comment]
Thanks for this @kevinmatthes, good practice to decrease permissions. Just a little edit in the `CITATION.cff` references, then it's good to go AFAIC.

**Comment by jspaaks** [comment]
Awesome, thanks again!

**Pull request merged**